main.c

```diff
@@ -9,21 +10,29 @@
 int** slave(int *set1, int *set2, int N, int NUM_SETS) {
 
   // Holds the 4 newly mutated sets
-  int **new_sets = (int **)malloc(NUM_SETS * sizeof(int *)); 
-  
+  int **new_sets = (int **)malloc(NUM_SETS * sizeof(int *));
+
   for (int i = 0; i < NUM_SETS; i++) {
-      new_sets[i] = (int *)malloc(N * N * sizeof(int));
+      new_sets[i] = (int *)malloc(N * sizeof(int));
   }
 
   // TODO: Incorporate the mutate functions
+  // TODO: Use OMP to improve performance
+
 
-  return new_sets; 
+  new_sets = crossoverRandomSplit(N, set1, set2, 0);
```

Comment:
I don't think you're using the mutation algorithms here correctly. You should be storing the new sets from crossOverRandomSplit into the first two indexes of `new_sets` and then the new sets from mutateMaxConflict into the last 2 indexes of `new_sets`

Reply:
resolved, also it's times like this when i'm very happy i chose the username i did